sentry_sdk/integrations/django/__init__.py

```diff
@@ -612,7 +612,13 @@ def execute(self, sql, params=None):
         with record_sql_queries(
             hub, self.cursor, sql, params, paramstyle="format", executemany=False
         ) as span:
-            _set_db_data(span, self.db.vendor, self.db.get_connection_params())
+            connection_parameters = (
+                self.connection.get_dsn_parameters()
+                if hasattr(self.connection, "get_dsn_parameters")
+                else self.db.get_connection_params()
+            )
```

Comment:
We could move this to `_set_db_data` and just have `_set_db_data` take the `span` and `self` as args to not repeat this multiple times